src/app/clusters/software-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -18,31 +18,41 @@
 #include <app/clusters/software-diagnostics-server/software-diagnostics-logic.h>
 #include <app/static-cluster-config/SoftwareDiagnostics.h>
 #include <data-model-providers/codegen/CodegenDataModelProvider.h>
+#include <lib/core/Global.h>
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::SoftwareDiagnostics;
+using chip::app::Clusters::SoftwareDiagnostics::StaticApplicationConfig::IsAttributeEnabledOnSomeEndpoint;
 
 // this file is ever only included IF software diagnostics is enabled and that MUST happen only on endpoint 0
 static_assert(SoftwareDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 1,
-              "Exactly one softare diagnistics provider may exist on Endpoint 0");
+              "Exactly one software diagnostics cluster instance may exist");
 static_assert(SoftwareDiagnostics::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId,
-              "Exactly one softare diagnistics provider may exist on Endpoint 0");
+              "The software diagnostics cluster must be on endpoint 0");
 
 namespace {
 
-RegisteredServerCluster<SoftwareDiagnosticsServerCluster<DeviceLayerSoftwareDiagnosticsLogic>> gServer;
+Global<LazyRegisteredServerCluster<SoftwareDiagnosticsServerCluster<DeviceLayerSoftwareDiagnosticsLogic>>> gServer;
```

Comment:
If we're using LazyRegisteredServerCluster do we even need Global?  LazyRegisteredServerCluster has a constexpr constructor...  The point of Global is to avoid static initializers.

Reply:
sure, I can drop global. I added the lazy only later because I needed a constructor argument for enabling bits.